### PR TITLE
Support a custom includes path

### PR DESCRIPTION
**Pull request by codekoala**
This adds support for a custom includes path. This helps when you do not have permission to modify `/usr/local/include`, for example.

It relies on the standard `-I` flag, so something like `make -I /tmp/custom/path` and `make -I/var/lib/whatever` work.

### CODE REVIEW

**Comment by codekoala** [comment]
@tj any thoughts on this? I have a project waiting on this feature for its build pipeline.

**Comment by tj** [comment]
hmm not something I need personally. I'm not a huge fan of replacing pkg-level globals from an API, but I think otherwise looks decent to me!

I kinda feel like using `-I` ruins any niceness that is provided by mmake but I'm not sure what your use-case is.

**Comment by codekoala** [comment]
Thank you for the feedback!

I'm trying to use `mmake` in a build pipeline that does not provide write access to the build system's `/usr/local/include` directory. Using the `-I ./include` or whatever allows me to still pull in all of the dependencies using `mmake`'s magic, but instead of writing those includes to `/usr/local/include` they're written to a location the build job has write access to.

File: resolver/resolver.go [hunk 1]
```diff
@@ -15,10 +19,33 @@ var (
...
 	// Get must return the contents of the file,
 	// ErrNotFound when the file is not found,
 	// or an error.
 	Get(path string) (io.ReadCloser, error)
 }
+
+// GetIncludePath allows the include path to be overridden.
+func GetIncludePath(args []string) string {
+	// always reset the global IncludePath to the default
+	IncludePath = DefaultIncludePath
```
Comment (tj) [inline resolver/resolver.go]: is this line necessary if it's defaulted above?

**Comment by tj** [comment]
SGTM! just one question about the one line but otherwise I'll merge!

File: resolver/resolver.go (hunk 1 above)
Comment (codekoala) [reply resolver/resolver.go]: That's a good question. I know I had a reason for doing it this way, but that reason escapes me at this moment. I'll have to take a closer look at the whole diff when I get back to a computer.

Comment (codekoala) [reply resolver/resolver.go]: I think I added this line so the test results would be consistent. Without it, the expected outcome for later tests would not match. This being the case, I'll just reset `IncludePath` in the test instead.

**Comment by codekoala** [comment]
@tj this should be good to go now. Thanks!

**Comment by codekoala** [comment]
@tj any other objections?

**Comment by zph** [comment]
Squashing and merging this and pushing a follow on commit to fix test reporting that changed. Thanks @codekoala!

**Comment by codekoala** [comment]
Thank you for merging!!!

**Comment by zph** [comment]
👍 💃

**Pull request merged**